tests/backup/backup_license_test.go

```diff
@@ -105,36 +110,208 @@ var _ = Describe("{NodeCountForLicensing}", func() {
+		})
+		Step("Verifying the license count after adding source and destination clusters with all worker nodes labelled portworx.io/nobackup=true", func() {
+			log.InfoD("Verifying the license count after adding source and destination clusters with all worker nodes labelled portworx.io/nobackup=true")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			err = VerifyLicenseConsumedCount(ctx, orgID, 0)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying that license count after adding source and destination clusters with all worker nodes labelled portworx.io/nobackup=true should be 0"))
+		})
+		Step("Taking backup of applications", func() {
+			log.InfoD("Taking Backup of application")
+			backupName = fmt.Sprintf("%s-%v", BackupNamePrefix, time.Now().Unix())
+			err = CreateBackup(backupName, SourceClusterName, bkpLocationName, backupLocationUID, bkpNamespaces,
+				nil, orgID, clusterUid, "", "", "", "", ctx)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Taking backup: %s", backupName))
+		})
+		Step("Restoring the backed up application", func() {
+			log.InfoD("Restoring the backed up application")
+			restoreName = fmt.Sprintf("%s-%v", RestoreNamePrefix, time.Now().Unix())
+			err = CreateRestore(restoreName, backupName, nil, destinationClusterName, orgID, ctx, nil)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Creating restore: %s of backup: %s", backupName, backupName))
+		})
+		Step("Removing label portworx.io/nobackup=true from worker nodes and verifying the license count", func() {
+			log.InfoD("Removing label from one worker node on source cluster and verifying the license count")
+			err := Inst().S.RemoveLabelOnNode(sourceClusterWorkerNodes[0], "portworx.io/nobackup")
+			log.FailOnError(err, fmt.Sprintf("Failed to remove label portworx.io/nobackup=true from worker node %v", sourceClusterWorkerNodes[0].Name))
+			err = VerifyLicenseConsumedCount(ctx, orgID, 1)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying license count after removing label portworx.io/nobackup=true from one worker node: %v should be 1", sourceClusterWorkerNodes[0].Name))
+			log.InfoD("Switching cluster context to destination cluster")
+			err = SetDestinationKubeConfig()
+			log.FailOnError(err, "Switching context to destination cluster failed")
+			log.InfoD("Removing label from all worker nodes on destination cluster and verifying the license count")
+			for _, workerNode := range destinationClusterWorkerNodes {
+				err = Inst().S.RemoveLabelOnNode(workerNode, "portworx.io/nobackup")
+				log.FailOnError(err, fmt.Sprintf("Failed to remove label portworx.io/nobackup=true from worker node %v", workerNode.Name))
+			}
+			log.InfoD("Switching cluster context back to source cluster")
+			err = SetSourceKubeConfig()
+			log.FailOnError(err, "Switching context to source cluster")
+			err = VerifyLicenseConsumedCount(ctx, orgID, int64(len(destinationClusterWorkerNodes))+1)
```

Comment:
Is it not sourceClusterWorkerNodes instead of destinationClusterWorkerNodes since we have switched the context to source cluster?

Reply:
As discussed we have to check the node count of destination cluster only here.

Reply:
Got it. Thanks!